Monika After Story/game/script-topics.rpy

```diff
@@ -1854,6 +1842,295 @@ label monika_rain_holdme:
         m 1dsc "Sorry..."
     return
 
+label monika_holdme_prep:
+    stop music fadeout 1.0
+
+    # clear selected track
+    $ songs.current_track = songs.FP_NO_SONG
+    $ songs.selected_track = songs.FP_NO_SONG
+    
+    # hide ui and disable hotkeys
+    $ HKBHideButtons()
+    $ store.songs.enabled = False
+    return
+
+label monika_holdme_start:
+    show monika 6dubsa
+    #Start the timer vv
+    $ start_time = datetime.datetime.now()
+
+    $ ui.add(PauseDisplayable())
+    $ ui.interact()
+
+    # renable ui and hotkeys
+    $ store.songs.enabled = True
+    $ HKBShowButtons()
+    return
+
+label monika_holdme_reactions:
+    if mas_isMoniLove():
```

Comment:
I would do time checking first, then aff checking. By doing time checking first, its easier to see what times elicit reactions. Additionally, time checking has to create a timedelta object for each conditional check, so having that be the outer layer is slightly more efficient.

Something like:
```python
if time is greater than 30 mins:
   if love:
   elif enamored:
   elif affectionate:
   # ...
elif time greater than 10 mins:
   if love:
   elif enamored:
   #...
else:
   # ...